server/api/impl/aws/provision_app.go

```diff
@@ -187,31 +220,21 @@ func (c *Client) createFunction(zipFile io.Reader, function, handler, runtime, r
 	return nil
 }
 
-// getFunctionName generates function name for a specific app
-// name can be 64 characters long.
-func getFunctionName(appID apps.AppID, version apps.AppVersion, function string) (string, error) {
-	if len(appID) > appIDLengthLimit {
-		return "", errors.Errorf("appID %s too long, should be %d bytes", appID, appIDLengthLimit)
+func (c *Client) provisionAssets(manifest *apps.Manifest, assets []assetData) (*apps.Manifest, error) {
+	if manifest.Assets == nil {
+		manifest.Assets = make([]apps.Asset, 0, len(assets))
 	}
-	if len(version) > len(versionFormat) {
-		return "", errors.Errorf("version %s too long, should be in %s format", version, versionFormat)
-	}
-
-	// Sanitized any dots used in appID and version as lambda function names can not contain dots
-	// While there are other non-valid characters, a dots is the most commonly used one
-	sanitizedAppID := strings.ReplaceAll(string(appID), ".", "-")
-	sanitizedVersion := strings.ReplaceAll(string(version), ".", "-")
-
-	name := fmt.Sprintf("%s_%s_%s", sanitizedAppID, sanitizedVersion, function)
-	if len(name) <= lambdaFunctionFileNameMaxSize {
-		return name, nil
-	}
-	functionNameLength := lambdaFunctionFileNameMaxSize - len(sanitizedAppID) - len(sanitizedVersion) - 2
-	hash := sha256.Sum256([]byte(name))
-	hashString := hex.EncodeToString(hash[:])
-	if len(hashString) > functionNameLength {
-		hashString = hashString[:functionNameLength]
-	}
-	name = fmt.Sprintf("%s-%s-%s", sanitizedAppID, sanitizedVersion, hashString)
-	return name, nil
+	for _, asset := range assets {
+		key := getAssetFileKey(manifest.AppID, manifest.Version, asset.name)
+		if err := c.S3FileUpload(key, asset.file); err != nil {
+			return nil, errors.Wrapf(err, "can't provision asset - %s with key - %s", asset.name, key)
+		}
+		manifest.Assets = append(manifest.Assets, apps.Asset{
+			Name:   asset.name,
+			Type:   apps.S3Asset,
+			Bucket: c.appsS3Bucket,
+			Key:    key,
```

Comment:
I'm wondering if we even need need all of that metadata for an Asset. 

1. Type: Is it safe to assume that an HTTP App only has HTTP assets and an AWS App only has S3 Assets? I don't see a case were we would mix that.
2. Key: Why do we need to store they key if we can generate it from the other data?
3. Why no we need to store the bucket, if that is fixed?